test/integration/generated/additionalProperties/src/additionalPropertiesClient.ts

```diff
@@ -46,6 +51,32 @@ export class AdditionalPropertiesClient extends coreClient.ServiceClient {
     // Assigning values to Constant parameters
     this.$host = options.$host || "http://localhost:3000";
     this.pets = new PetsImpl(this);
+    if (options.apiVersion) {
+      this.customApiVersion(options.apiVersion);
+    }
+  }
+
+  /**  A policy that sets the api-version (or equivalent) to reflect the library version. */
+  private customApiVersion(apiVersion: string) {
```

Comment:
nit: you could also make this `apiVersion?: string` and move your `if(options.apiVersion)` check inside this function and bail if no apiVersion is provided. With this, you'll keep the constructor simpler